src/lib.rs

```diff
@@ -1,103 +1,197 @@
+///
+/// # Usage
+///
+/// This module is intended to be used in conjunction with the [`HorizonClient`](crate::horizon_client::HorizonClient)
+/// for making specific account-related API calls to the Horizon server. The request
+/// structures are designed to be passed to the client's methods, which handle the
+/// communication with the server and return the corresponding response structures.
+///
+/// # Example
+/// An example of retrieving a list of accounts, filtering by signer:
+/// ```rust
+/// # use stellar_rust_sdk::accounts::prelude::{AccountsRequest, AccountsResponse};
+/// # use stellar_rust_sdk::models::Request;
+/// # use stellar_rust_sdk::horizon_client::HorizonClient;
+/// #
+/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+/// # let base_url = "https://horizon-testnet.stellar.org".to_string();
+/// # let horizon_client = HorizonClient::new(base_url)
+/// #    .expect("Failed to create Horizon Client");
+/// let request = AccountsRequest::new()
+///     .set_signer("GDQJUTQYK2MQX2VGDR2FYWLIYAQIEGXTQVTFEMGH2BEWFG4BRUY4CKI7").unwrap()
+///     .set_limit(10).unwrap();
+///
+/// let response: Result<AccountsResponse, String> = horizon_client
+///     .get_account_list(&request)
+///     .await;
+/// # Ok({})
+/// # }
+///
 pub mod accounts;
-/// The assets module
+
 pub mod assets;
-/// The claimable balances module
+
 pub mod claimable_balances;
-/// The horizon client module
+
+/// Client for calling the Stellar Horizon API
+///
+/// # Constructing a `HorizonClient`
+/// A string containing the base URL for the Horizon API is required to contruct a client.
+/// For example, to construct a client for the Horizon API testnet:
+/// ```rust
+/// use stellar_rust_sdk::horizon_client::HorizonClient;
+///
+/// let base_url = "https://horizon-testnet.stellar.org".to_string();
+/// let horizon_client = HorizonClient::new(base_url)
+///     .expect("Failed to create Horizon Client");;
+/// ```
+///
+/// # Using the `HorizonClient`
+/// The HorizonClient has a function that can be called for each endpoind provided
+/// by the Horizon API. For example, it has a [`HorizonClient::get_account_list`](crate::horizon_client::HorizonClient::get_account_list)
+/// function, which returns an async future that contains a result, as illustrated below:
+/// ```rust
+/// # use stellar_rust_sdk::assets::prelude::{AllAssetsRequest, AllAssetsResponse};
+/// # use stellar_rust_sdk::models::Request;
+/// # use stellar_rust_sdk::horizon_client::HorizonClient;
+/// #
+/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+/// # let base_url = "https://horizon-testnet.stellar.org".to_string();
+/// # let horizon_client = HorizonClient::new(base_url)
+/// #    .expect("Failed to create Horizon Client");;
+/// let all_assets_request = AllAssetsRequest::new();
+/// let accounts_response: Result<AllAssetsResponse, String> = horizon_client
+///     .get_all_assets(&all_assets_request)
+///     .await;
+/// # Ok({})
+/// # }
```

Comment:
This also has been documentated at HorizonClient::getAllAssets. The module itself provides more functionality, so it should not be documented here for a specific endpoint I believe.